tests/test_flagging_grouping.py

```diff
@@ -0,0 +1,106 @@
+"""Tests for grouping and flagging"""
+import logging
+import os
+import unittest
+import pytest
+import copy
+
+import astropy.units as u
+import numpy as np
+
+import pint.models.model_builder as mb
+import pint.toa as toa
+from pint.residuals import Residuals
+from pinttestdata import datadir
+from pint.models import parameter as p
+from pint.models import PhaseJump
+
+
+class SimpleSetup:
+    def __init__(self, par, tim):
+        self.par = par
+        self.tim = tim
+        self.m = mb.get_model(self.par)
+        self.t = toa.get_TOAs(
+            self.tim, ephem="DE405", planets=False, include_bipm=False
+        )
+
+
+@pytest.fixture
+def setup_NGC6440E():
+    os.chdir(datadir)
+    return SimpleSetup("NGC6440E.par", "NGC6440E.tim")
+
+
+def test_add_flags(setup_NGC6440E):
+    """
+    test adding a new flag
```

Comment:
Docstrings in test functions are not displayed anywhere but their names are; please put the same information in the name (no one is calling these functions, so there's no problem if they're long to type).